Module Does Not Compile with LibreSSL 3.0+

Compile error - can not get nginx to compile due to this error. 

ngx_http_lua_ssl_certby.c:1452:13: error: implicit declaration of function ‘SSL_set0_verify_cert_store’; did you mean ‘SSL_set_verify_result’?

It seems lua-nginx-module-0.10.19 is not building with LibreSSL. I have tried LibreSSL 3.2 and LibreSSL 3.3. This issue is new to 0.10.19 and does not occur if I drop back to lua-nginx-module-0.10.17 with LibreSSL

some basic info from my compile script

LUA_JIT_VER=2.1-20201027
LUA_NGINX_VER=0.10.19
LUA_RESTY_CORE=0.1.21
LIBRESSL_VER=3.3.0
NGINX_MAINLINE_VER=1.19.5
NGINX_DEV_KIT=0.3.1 

It seems there is no function definition found - maybe the order of declaration? I am not a C programmer so I don't understand the error. Some stack trace below.
`cc -c -I/usr/local/include/luajit-2.1/  -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -Wno-ignored-qualifiers  -D_GLIBCXX_USE_CXX11_ABI=0 -Wno-deprecated-declarations -DNDK_SET_VAR  -I src/core -I src/event -I src/event/modules -I src/os/unix -I /usr/local/src/nginx/modules/ngx_devel_kit-0.3.1/objs -I objs/addon/ndk -I /usr/local/src/nginx/modules/lua-nginx-module-0.10.19/src/api -I /usr/local/src/nginx/modules/libressl-3.3.0/.openssl/include -I obj
        -o objs/addon/src/ngx_http_lua_ssl_ocsp.o \
        /usr/local/src/nginx/modules/lua-nginx-module-0.10.19/src/ngx_http_lua_ssl_ocsp.c
/usr/local/src/nginx/modules/lua-nginx-module-0.10.19/src/ngx_http_lua_ssl_certby.c: In function ‘ngx_http_lua_ffi_ssl_verify_client’:
/usr/local/src/nginx/modules/lua-nginx-module-0.10.19/src/ngx_http_lua_ssl_certby.c:1452:13: error: implicit declaration of function ‘SSL_set0_verify_cert_store’; did you mean ‘SSL_set_verify_result’? [-Werror=implicit-function-declaration]
 1452 |         if (SSL_set0_verify_cert_store(ssl_conn, ca_store) == 0) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |             SSL_set_verify_result
cc -c -I/usr/local/include/luajit-2.1/  -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -Wno-ignored-qualifiers  -D_GLIBCXX_USE_CXX11_ABI=0 -Wno-deprecated-declarations -DNDK_SET_VAR  -I src/core -I src/event -I src/event/modules -I src/os/unix -I /usr/local/src/nginx/modules/ngx_devel_kit-0.3.1/objs -I objs/addon/ndk -I /usr/local/src/nginx/modules/lua-nginx-module-0.10.19/src/api -I /usr/local/src/nginx/modules/libressl-3.3.0/.openssl/include -I obj>
        -o objs/addon/src/ngx_http_lua_lex.o \
        /usr/local/src/nginx/modules/lua-nginx-module-0.10.19/src/ngx_http_lua_lex.c
cc1: all warnings being treated as errors
make[1]: *** [objs/Makefile:2238: objs/addon/src/ngx_http_lua_ssl_certby.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/usr/local/src/nginx/nginx-1.19.5'
make: *** [Makefile:8: build] Error 2
`

--- a/src/ngx_http_lua_ssl_certby.c
+++ b/src/ngx_http_lua_ssl_certby.c
@@ -1350,6 +1350,13 @@ int
 ngx_http_lua_ffi_ssl_verify_client(ngx_http_request_t *r, void *ca_certs,
     int depth, char **err)
 {
+#ifdef LIBRESSL_VERSION_NUMBER
+
+    *err = "LibreSSL not supported";
+    return NGX_ERROR;
+
+#else
+
     ngx_http_lua_ctx_t          *ctx;
     ngx_ssl_conn_t              *ssl_conn;
     ngx_http_ssl_srv_conf_t     *sscf;
@@ -1466,6 +1473,7 @@ failed:
     X509_STORE_free(ca_store);
 
     return NGX_ERROR;
+#endif
 }
 
 

